back/src/main/java/ro/utcn/sd/he/assignment1/controller/QuestionsController.java

```diff
@@ -1,55 +1,98 @@
 package ro.utcn.sd.he.assignment1.controller;
 
 import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.context.event.EventListener;
+import org.springframework.messaging.simp.SimpMessagingTemplate;
 import org.springframework.web.bind.annotation.*;
 import ro.utcn.sd.he.assignment1.command.AddQuestionCommand;
 import ro.utcn.sd.he.assignment1.command.Command;
 import ro.utcn.sd.he.assignment1.dto.QuestionDTO;
+import ro.utcn.sd.he.assignment1.dto.QuestionVoteDTO;
 import ro.utcn.sd.he.assignment1.dto.UserDTO;
+import ro.utcn.sd.he.assignment1.dto.VoteDTO;
+import ro.utcn.sd.he.assignment1.event.BaseEvent;
+import ro.utcn.sd.he.assignment1.event.QuestionVoteEvent;
 import ro.utcn.sd.he.assignment1.model.Question;
+import ro.utcn.sd.he.assignment1.model.User;
 import ro.utcn.sd.he.assignment1.service.QuestionService;
 import ro.utcn.sd.he.assignment1.service.QuestionTagService;
+import ro.utcn.sd.he.assignment1.service.UserService;
+import ro.utcn.sd.he.assignment1.service.VoteService;
 
 import java.util.ArrayList;
 import java.util.List;
 
+@Slf4j
 @RequiredArgsConstructor
 @RestController
 public class QuestionsController {
     private final QuestionService questionService;
     private final QuestionTagService questionTagService;
+    private final VoteService voteService;
+    private final UserService userService;
+    private final SimpMessagingTemplate messagingTemplate;
     private List<Command> commands = new ArrayList<>();
+
     @GetMapping("questions/{id}")
     public UserDTO readSingle(@PathVariable int id) {
         //TODO
         return null;
     }
 
     @GetMapping("/questions")
-    public List<Question> readAll() {
+    public List<QuestionVoteDTO> readAll() {
         return questionService.listQuestions();
     }
 
     @GetMapping("/questions/title/{title}")
-    public List<Question> searchTitle(@PathVariable String title){
+    public List<QuestionVoteDTO> searchTitle(@PathVariable String title) {
         return questionService.searchTitle(title);
     }
 
     @GetMapping("/questions/tag/{tagName}")
-    public List<Question> searchTag(@PathVariable String tagName){
+    public List<QuestionVoteDTO> searchTag(@PathVariable String tagName) {
         return questionTagService.filterTag(tagName);
     }
 
+    /*
+    is it the eventPublisher that we shouldn't depend on in order to maintain the design?
+    or the event? if neither of those is the case, why can't we just make it directly here?
+     */
```

Comment:
Yes, we could just send WebSocket events via the Simp template directly in the controller, without having to rely on event publishers. But the event itself is more part of the business logic than the controller layer (in practice you could have several listeners - for example maybe you have in-app notifications like facebook and you want to persist them in the DB - in this case , it is easy to implement it if you already use the event publisher).